Fix compiler warnings on 16-bit architecture

henriknil opened the pull request:
On archtectures where unsigned int is only 16 bits the warning:
"unsigned conversion from 'long int' to 'unsigned int' changes
value from 'xxx' to 'yyy' [-Werror=overflow] and "left shift count
>= width of type [-Werror=shift-count-overflow]" where issued.

henriknil committed:
adler32: Fix compiler warning on 16-bit architecture

On archtectures where unsigned int is only 16 bits the warning:
"left shift count >= width of type [-Werror=shift-count-overflow]"
was issued.

pfalcon commented:
Thanks, but please follow commit style of the project (git log): commit message format, and not directly related changes should be separate.

henriknil committed:
crc32: Fix compiler warnings on 16-bit architecture

On archtectures where unsigned int is only 16 bits the warning:
"unsigned conversion from 'long int' to 'unsigned int' changes
value from 'xxx' to 'yyy' [-Werror=overflow] was issued.

pfalcon commented:
Thanks for the updates, merged.
